src/HDF5.jl

```diff
@@ -704,18 +704,20 @@ function split1(path::AbstractString)
 end
 
 function create_group(parent::Union{File,Group}, path::AbstractString,
-                  lcpl::Properties=_link_properties(path),
-                  gcpl::Properties=DEFAULT_PROPERTIES)
+                  lcpl::LinkCreateProperties=_link_properties(path),
+                  gcpl::GroupCreateProperties=GroupCreateProperties())
     haskey(parent, path) && error("cannot create group: object \"", path, "\" already exists at ", name(parent))
     Group(API.h5g_create(parent, path, lcpl, gcpl, API.H5P_DEFAULT), file(parent))
 end
 
 # Setting dset creation properties with name/value pairs
 function create_dataset(parent::Union{File,Group}, path::AbstractString, dtype::Datatype, dspace::Dataspace; pv...)
-    dcpl = isempty(pv) ? DEFAULT_PROPERTIES : create_property(API.H5P_DATASET_CREATE; pv...)
-    dxpl = isempty(pv) ? DEFAULT_PROPERTIES : create_property(API.H5P_DATASET_XFER; pv...)
-    dapl = isempty(pv) ? DEFAULT_PROPERTIES : create_property(API.H5P_DATASET_ACCESS; pv...)
     haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
```

Comment:
I moved this up since, it would make sense to error before constructing any properties objects. Hope that's ok with you.